### PR TITLE
Issue 1620

### PR DESCRIPTION
**Pull request by jkburges**
This does not solve https://github.com/aodn/aodn-portal/issues/1620, but should enable us to provide some better feedback to the user, which *will* hopefully resolve it.

This PR adds some events to indicate a download's status.  Currently, the only 'action' is to log a message when one of the new events is fired (see https://github.com/aodn/aodn-portal/blob/issue_1620/web-app/js/portal/cart/DownloadPanelBody.js#L122) but this could easily be extended to give some appropriate feedback to the user).

### CODE REVIEW

**Comment by anguss00** [comment]
I'll review this

**Comment by jkburges** [comment]
Ok, but perhaps don't merge yet @anguss00 until we have our little meeting at 2pm...

**Comment by anguss00** [comment]
Yep, sure. I'm just more keen to check out what's changing through here

File: src/groovy/au/org/emii/portal/proxying/RequestProxyingController.groovy [hunk 1]
```diff
@@ -33,17 +35,32 @@ abstract class RequestProxyingController {
             render text: "Host for address '$url' not allowed", contentType: "text/html", encoding: "UTF-8", status: HTTP_400_BAD_REQUEST
```
Comment (anguss00) [inline src/groovy/au/org/emii/portal/proxying/RequestProxyingController.groovy]: I think if not allowed, we should call this is 403 forbidden error

**Comment by jkburges** [comment]
@anguss00
image

**Comment by anguss00** [comment]
:+1:

File: grails-app/controllers/au/org/emii/portal/DownloadController.groovy [hunk 2]
```diff
@@ -23,6 +23,12 @@ class DownloadController extends RequestProxyingController {
     static def SIZE_ESTIMATE_ERROR = "-1"
 
     // Index action inherited from RequestProxyingController
+    def index = {
+        Thread.sleep(5000)
```
Comment (anguss00) [inline grails-app/controllers/au/org/emii/portal/DownloadController.groovy]: thread sleep still here ;-)

Comment (jkburges) [inline grails-app/controllers/au/org/emii/portal/DownloadController.groovy]: screwed up rebase...

Comment (jkburges) [inline grails-app/controllers/au/org/emii/portal/DownloadController.groovy]: ok fixed that one.

File: web-app/js/portal/cart/DownloadPanelBody.js [hunk 3]
```diff
@@ -117,7 +117,14 @@ Portal.cart.DownloadPanelBody = Ext.extend(Ext.Panel, {
     confirmDownload: function(collection, generateUrlCallbackScope, generateUrlCallback, params, textKey) {
 
         params.onAccept = function(callbackParams) {
-            var downloader = new Portal.cart.Downloader();
+            var downloader = new Portal.cart.Downloader({
+                listeners: {
+                    'downloadrequested': function(downloadUrl) { log.debug('Download requested', downloadUrl); },
+                    'downloadstarted': function(downloadUrl) { log.debug('Download started', downloadUrl); },
+                    'downloadfailed': function(downloadUrl, msg) { log.debug('Download failed', downloadUrl, msg); },
```
Comment (anguss00) [inline web-app/js/portal/cart/DownloadPanelBody.js]: extra comma here

Comment (jkburges) [inline web-app/js/portal/cart/DownloadPanelBody.js]: image

**Comment by anguss00** [comment]
Alrighty, looks great @jkburges

**Comment by jkburges** [comment]
image

**Comment by anguss00** [comment]
fist-bump

**Pull request merged**